test/com/facebook/buck/jvm/java/JavaOrderingIntegrationTest.java

```diff
@@ -0,0 +1,135 @@
+/*
+ * Copyright 2018-present Facebook, Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License"); you may
+ * not use this file except in compliance with the License. You may obtain
+ * a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+ * WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the
+ * License for the specific language governing permissions and limitations
+ * under the License.
+ */
+
+package com.facebook.buck.jvm.java;
+
+import com.facebook.buck.jvm.java.testutil.AbiCompilationModeTest;
+import com.facebook.buck.testutil.ProcessResult;
+import com.facebook.buck.testutil.TemporaryPaths;
+import com.facebook.buck.testutil.integration.ProjectWorkspace;
+import com.facebook.buck.testutil.integration.TestDataHelper;
+import com.google.common.base.Joiner;
+import java.io.IOException;
+import java.io.InputStream;
+import java.io.InputStreamReader;
+import java.nio.file.Path;
+import java.util.Arrays;
+import java.util.zip.ZipFile;
+import org.junit.Assert;
+import org.junit.Rule;
+import org.junit.Test;
+
+/**
+ * See https://github.com/facebook/buck/issues/1830 for background.
+ *
+ * Integration test to verify:
+ * 1. Java tests and java binaries access parameters in the same order.
+ * 2. Order of the resolution.
+ * Changing resolution order is breaking change and this test will catch it.
+ */
+public class JavaOrderingIntegrationTest extends AbiCompilationModeTest {
```

Comment:
Could you split this class and move the tests to `JavaTestIntegrationTest` and `JavaBinaryTest`?

Reply:
I intentionally had all of these in the same test so it is clear that these are related and have to be in the same ordering. I can split it across specific tests if you still think it is more clean.